build_tools/azure/install.sh

```diff
@@ -120,6 +135,24 @@ python_environment_install() {
         pip install https://github.com/joblib/joblib/archive/master.zip
         echo "Installing pillow master"
         pip install https://github.com/python-pillow/Pillow/archive/main.zip
+    elif [[ "$DISTRIB" == "pip-nogil" ]]; then
+        setup_ccache  # speed-up the build of CPython it-self
```

Comment:
You can move `setup_ccache` earlier e.g. in `pre_python_environment_install` and remove it from `scikit_learn_install` this way it will always be called once. 

Not sure whether then you'd want the "ccache already configured, skipping..." logic

Other than this LGTM,

Reply:
> You can move setup_ccache earlier e.g. in pre_python_environment_install and remove it from scikit_learn_install this way it will always be called once.

This is what I wanted to do initially but it does not work because on some builds (in particular the macos builds) the ccache command is installed with conda instead of apt.